File: Semestrul 3/MAP/lab 2 MAP/model/StackContainer.java
package sem1_2.model;

import java.util.ArrayList;
import java.util.List;
import sem1_2.model.AbstractContainer;

public class StackContainer extends AbstractContainer {

    List<Task> tasks = new ArrayList<>();

    @Override
    public Task remove() {
        if (!tasks.isEmpty()) {
            return tasks.remove(tasks.size() - 1);
        } else {
            return null;
        }
    }

    @Override
    public void add(Task task) {
        tasks.add(task);
    }
}

File: Semestrul 3/MAP/lab 10 MAP/model/StackContainer.cs
namespace seminar10.model;

public class StackContainer : AbstractContainer
{
    public override Task Remove()
    {
        if (_tasks.Count == 0) return null!;
        var lastTask = _tasks[^1];
        _tasks.RemoveAt(_tasks.Count - 1);
        return lastTask;
    }

    public override void Add(Task task)
    {
        _tasks.Add(task);
    }
}